### PR TITLE
remove incorrect usages of Optional class in parser

### PR DESCRIPTION
**Pull request by chaudum**
Replace Optional usages with Nullable properties, because they are only used for "null checks".

### CODE REVIEW

File: sql-parser/src/main/java/io/crate/sql/tree/AliasedRelation.java [hunk 1]
```diff
@@ -82,7 +82,7 @@ public boolean equals(Object o) {
         if (!alias.equals(that.alias)) {
             return false;
         }
-        if (columnNames != null ? !columnNames.equals(that.columnNames) : that.columnNames != null) {
+        if (!columnNames.equals(that.columnNames)) {
```
Comment (mxm) [inline sql-parser/src/main/java/io/crate/sql/tree/AliasedRelation.java]: Could you add a `checkNotNull` check in the constructor above like for the other fields?

File: sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java [hunk 2]
```diff
@@ -943,13 +943,13 @@ public Node visitQuotedIdentifierAlternative(SqlBaseParser.QuotedIdentifierAlter
         return new StringLiteral(context.getText().replace("\"\"", "\""));
     }
 
+    @Nullable
     private String getIdentText(SqlBaseParser.IdentContext ident) {
```
Comment (mfussenegger) [inline sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java]: also annotate the argument as nullable

File: sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java [hunk 3]
```diff
@@ -1481,6 +1481,14 @@ protected Node aggregateResult(Node aggregate, Node nextResult) {
         throw new UnsupportedOperationException("not yet implemented");
     }
 
+    @Nullable
+    private <T> T visitOptionalContext(ParserRuleContext context, Class<T> clazz) {
```
Comment (mfussenegger) [inline sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java]: also annotate the argument as nullable

File: sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java [hunk 4]
```diff
@@ -1517,13 +1525,14 @@ private static boolean isDistinct(SqlBaseParser.SetQuantContext setQuantifier) {
         return setQuantifier != null && setQuantifier.DISTINCT() != null;
     }
 
-    private static Optional<String> getTextIfPresent(ParserRuleContext context) {
-        return Optional.ofNullable(context).map(ParseTree::getText);
+    @Nullable
+    private static String getUnquotedText(ParserRuleContext context) {
```
Comment (mfussenegger) [inline sql-parser/src/main/java/io/crate/sql/parser/AstBuilder.java]: also annotate the argument as nullable

**Pull request merged**